src/libs/actions/Transaction.ts

```diff
@@ -572,6 +601,258 @@ function getAllTransactions() {
+                },
+                {
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+                    value: {[newIOUAction.reportActionID]: {childReportID: null}},
+                },
+            );
+        }
+
+        // 6. Add MOVEDTRANSACTION or UNREPORTEDTRANSACTION report actions
+        const movedAction =
+            reportID === CONST.REPORT.UNREPORTED_REPORTID
+                ? buildOptimisticUnreportedTransactionAction(transactionThreadReportID, transaction.reportID)
+                : buildOptimisticMovedTransactionAction(transactionThreadReportID, reportID);
+
+        optimisticData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${transactionThreadReportID}`,
+            value: {[movedAction?.reportActionID]: movedAction},
+        });
+
+        successData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${transactionThreadReportID}`,
+            value: {[movedAction?.reportActionID]: {pendingAction: null}},
+        });
+
+        failureData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${transactionThreadReportID}`,
+            value: {[movedAction?.reportActionID]: null},
+        });
+
+        transactionIDToReportActionAndThreadData[transaction.transactionID] = {
+            movedReportActionID: movedAction.reportActionID,
+            moneyRequestPreviewReportActionID: newIOUAction.reportActionID,
+            ...(oldIOUAction.childReportID
+                ? {}
+                : {
+                      transactionThreadReportID,
+                      transactionThreadCreatedReportActionID,
+                  }),
+        };
+    });
+
+    // 7. Update the report totals
+    Object.entries(updatedReportTotals).forEach(([reportIDToUpdate, total]) => {
+        optimisticData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportIDToUpdate}`,
+            value: {total},
+        });
+
+        failureData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportIDToUpdate}`,
+            value: {total: allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportIDToUpdate}`]?.total},
+        });
+    });
+
+    const parameters: ChangeTransactionsReportParams = {
+        transactionList: transactionIDs.join(','),
+        reportID,
+        transactionIDToReportActionAndThreadData: JSON.stringify(transactionIDToReportActionAndThreadData),
+    };
+    if (isEditing) {
+        API.write(WRITE_COMMANDS.CHANGE_TRANSACTIONS_REPORT, parameters, {
+            optimisticData,
+            successData,
+            failureData,
+        });
+    } else {
+        Onyx.update(optimisticData);
+    }
```

Comment:
This doesn't look correct. If we're in the creation flow, we only merge Onyx changes to create transaction draft, e.g. 
https://github.com/Expensify/App/blob/85040e5116ac4a17ff405bb96a34b66c2280a807/src/pages/iou/request/step/IOURequestStepAttendees.tsx#L53-L58